datastore/database_pg.go

```diff
@@ -0,0 +1,46 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ * License granted by Canonical Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datastore
+
+import (
+	"database/sql"
+	"log"
+
+	_ "github.com/lib/pq" // postgresql driver
+)
+
+// openPostgreSQLDatabase return an open database connection for an postgresql database
+func openPostgreSQLDatabase(driver, dataSource string) {
+	// Open the database connection
+	db, err := sql.Open(driver, dataSource)
+	if err != nil {
```

Comment:
There is a common code in both drivers opening. I wonder if that could be in a common file/site

Reply:
I did wonder that, but thought it would be safer to have the repeat, since there is some magic happening with the `_` import